MapleServer2/PacketHandlers/Game/GuildHandler.cs

```diff
@@ -541,6 +547,35 @@ private static void HandleListGuild(GameSession session, PacketReader packet)
         session.Send(GuildPacket.ListGuildUpdate(session.Player, toggle));
     }
 
+    private static void HandleGuildMail(GameSession session, IPacketReader packet)
+    {
+        string title = packet.ReadUnicodeString();
+        string body = packet.ReadUnicodeString();
+
+        Player sender = session.Player;
+        Guild guild = session.Player.Guild;
+
+        byte senderRank = sender.GuildMember.Rank;
+        GuildRank guildRank = guild.Ranks[senderRank];
+
+        if (!guildRank.HasRight(GuildRights.CanGuildMail))
+        {
+            session.Send(GuildPacket.ErrorNotice((byte)GuildErrorNotice.InsufficientPermissions));
+            return;
+        }
+
+        IEnumerable<long> guildMemberCharacterIds = guild.Members
+            .Select(m => m.Player.CharacterId)
+            .Where(i => i != sender.CharacterId);
+        
+        foreach (long characterId in guildMemberCharacterIds)
+        {
+            MailHelper.SendMail(MailType.Player, characterId, sender.CharacterId, sender.Name, title, body, "", "", new(), 0, 0, out Mail mail);
+
+            session.Send(MailPacket.Send(mail));
```

Comment:
it's meant to send the same packet several times to the sender?

Reply:
Maybe I've misunderstood the point of the MailPacket? From the wording, I understand it to be some sort of confirmation that the client is waiting for to confirm that the mail request finished being processed.

Is it perhaps the packet that drives the "You've got mail" notification that pops up? If this is the case, I believe there's a bug in the regular MailHandler too, I admit this was a lazy copy paste from [MailHandler::HandleSend:122](https://github.com/AlanMorel/MapleServer2/blob/master/MapleServer2/PacketHandlers/Game/MailHandler.cs#L122)

Reply:
I think you should ask for how it's supposed to work in discord since we can test it in the official server and also sniff to see what exactly the server sends.

Reply:
This should be good to go now